xdsl/traits.py

```diff
@@ -203,30 +203,51 @@ class SymbolOpInterface(OpTrait):
     a `SymbolTable` (TODO). A Symbol operation should use the SymbolOpInterface interface to
     provide the necessary verification and accessors.
 
-    Currently the only requirement is a "sym_name" attribute of type StringAttr.
+    A Symbol operation may be optional or not. If - the dault - it is not optional,
+    a `sym_name` attribute of type StringAttr is required. If it is optional,
+    the attribute is optional too.
 
-    Please see MLIR documentation for Symbol and SymbolTable for the requirements that are
-    upcoming in xDSL.
+    xDSL offers OptionalSymbolOpInterface as an always-optional SymbolOpInterface helper.
+
+    More requirements are defined in MLIR; Please see MLIR documentation for Symbol and
+    SymbolTable for the requirements that are upcoming in xDSL.
 
     https://mlir.llvm.org/docs/SymbolsAndSymbolTables/#symbol
     """
 
     @staticmethod
-    def get_sym_attr_name(op: Operation) -> StringAttr:
+    def get_sym_attr_name(op: Operation) -> StringAttr | None:
```

Comment:
```suggestion
    def get_sym_attr_name(self, op: Operation) -> StringAttr | None:
```
I think that this design is better in general for traits.
This allows to not have to fetch the trait twice.
Most of the time, this will be used in a code like this:

Reply:
Oh, that makes sense! I didn't know how to handle that case cleanly and I like this style 🙂